libfeed/feed-store-provider.c

```diff
@@ -0,0 +1,1074 @@
+append_discovery_feed_content_from_proxy (EosDiscoveryFeedKnowledgeAppProxy  *ka_proxy,
+                                          gpointer                            proxy_data,
+                                          GCancellable                       *cancellable,
+                                          GError                            **error)
+{
+  g_autoptr(AppendDiscoveryFeedContentFromProxyData) data = proxy_data;
+  g_autoptr(ArticleCardsFromShardsAndItemsData) marshal_data = article_cards_from_shards_and_items_data_new (ka_proxy,
+                                                                                                             data->direction,
+                                                                                                             data->type,
+                                                                                                             data->thumbnail_size,
+                                                                                                             data->factory);
+  GDBusProxy *dbus_proxy = eos_discovery_feed_knowledge_app_proxy_get_dbus_proxy (ka_proxy);
+  const gchar *desktop_id = eos_discovery_feed_knowledge_app_proxy_get_desktop_id (ka_proxy);
+
+  return call_dbus_proxy_and_construct_from_models_and_shards (dbus_proxy,
+                                                               data->method,
+                                                               desktop_id,
+                                                               article_cards_from_shards_and_items,
+                                                               marshal_data,
+                                                               cancellable,
+                                                               error);
+}
+
+static gboolean
+parse_int64_with_limits (const gchar  *str,
+                         guint         base,
+                         gint64        min,
+                         guint64       max,
+                         gint64       *out,
+                         GError      **error)
+{
+  gint64 ret = 0;
+
+  g_return_val_if_fail (out != NULL, FALSE);
+
+  /* Clear errno first before calling g_ascii_stroll */
+  errno = 0;
+
+  ret = g_ascii_strtoll (str, NULL, base);
+
+  if (errno != 0)
+    {
+      g_set_error (error,
+                   G_IO_ERROR,
+                   g_io_error_from_errno (errno),
+                   "Parsing of integer failed with %s", strerror (errno));
+      return FALSE;
+    }
+
+  if (ret < min || ret > max)
+    {
+      g_set_error (error,
+                   G_IO_ERROR,
+                   G_IO_ERROR_FAILED,
+                   "Parsing of integer failed, %lli is not "
+                   "in the range of %lli - %lli",
+                   ret,
+                   min,
+                   max);
+      return FALSE;
+    }
+
+  *out = ret;
+  return TRUE;
+}
+
+static gint64
+parse_base10_int64_with_limits (const gchar  *str,
```

Comment:
Feels odd to have so much indirection here; can't you call the inner function directly from the only caller of this?

Reply:
Yeah, fair enough. I changed it to call the inner function directly.